workflows/fetchdatairidanext.nf

```diff
@@ -56,10 +56,19 @@ workflow FETCHDATAIRIDANEXT {
     // Create a new channel of metadata from a sample sheet
     // NB: `input` corresponds to `params.input` and associated sample sheet schema
     input = Channel.fromSamplesheet("input")
-    meta_accessions = input.map {meta -> tuple(["id": meta.id.first()], meta.insdc_accession.first())}
+    // and remove non-alphanumeric characters in sample_names (meta.id)
+        .map { meta ->
+            if (meta.id[0]) {
+                    // Non-alphanumeric characters (excluding _,-,.) will be replaced with "_"
+                new_id = meta.id[0].replaceAll(/[^A-Za-z0-9_\.\-]/, '_') // meta.id appears to be an immutable list, the workaround is to create a new variable
+            } else {
+                new_id = meta.id[0]
+            }
+            return [["id": new_id, "irida_id": meta.irida_id[0], "insdc_accession": meta.insdc_accession[0]], meta.insdc_accession[0]]
+        }
```

Comment:
Not sure if this would work but could you reconstruct the entire object like this:

meta_accessions = input.map { meta ->
    def new_id = meta.id[0]?.replaceAll(/[^A-Za-z0-9_\.\-]/, '_') ?: meta.id[0]
    return [
        ["id": new_id, "irida_id": meta.irida_id[0], "insdc_accession": meta.insdc_accession[0]],
        meta.insdc_accession[0]
    ]
}

Reply:
Works, and looks good (better)! 02ecdb5